### ISSUE TITLE
🐛 BUG: `fetch` behaves differently on Cloudflare Pages and local preview

### ISSUE THREAD

**Issue by TangRufus** [cloudflare/wrangler2#748]
### What version of `Wrangler` are you using?

0.0.24

### What operating system are you using?

Mac

### Describe the Bug

`fetch` behaves differently on Cloudflare Pages and local preview (`wrangler pages dev`).

#### Steps to reproduce

1. Create `functions/demo.js`
	  ```js
	  // functions/demo.js
	  export const onRequestGet = async () => {
	    try {
	      await fetch('https://wordpress.org/')
	      return new Response('Fetch finished')
	    } catch (err) {
	      return new Response(JSON.stringify({
	        name: err.name,
	        message: err.message,
	        cause: err.cause,
	        stack: err.stack
	      }))
	    }
	  }
	  ```
1. Preview via `wrangler pages dev`. (Files in `./out` folder don't matter)
	```bash
	npx wrangler pages dev ./out
	````
1. Get the `/demo` page from local preview
    ```sh-session
    $ curl http://localhost:8788/demo
	{"name":"TypeError","message":"fetch failed","cause":{"code":"ERR_INVALID_CHAR"},"stack":"TypeError: fetch failed\n    at Object.processResponse (/Users/me/Code/my-demo/node_modules/undici/lib/fetch/index.js:183:23)\n    at Fetch.fetchFinale (/Users/me/Code/my-demo/node_modules/undici/lib/fetch/index.js:937:17)\n    at Fetch.mainFetch (/Users/me/Code/my-demo/node_modules/undici/lib/fetch/index.js:744:17)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)"}
    ```
1. Deploy to Cloudflare Pages
1. Get the `/demo` page from Cloudflare Pages
    ```sh-session
    $ curl https://xxx.my-page.pages.dev/demo
    Fetch finished
    ````
  
  #### Expected Behaviour 
  
  Steps 3 and 5 yield the same result.

**Comment by Skye-31** [comment]
This is an undici bug, which will be fixed when #1520 lands

**Comment by wiedymi** [comment]
I have same issue, how to solve it?

```wrangler 2.1.12```
